Source/CombatExtended/CombatExtended/Projectiles/BulletCE.cs

```diff
@@ -47,10 +49,12 @@ protected override void Impact(Thing hitThing)
                 // launcher being the pawn equipping the weapon, not the weapon itself
                 int damageAmountBase = def.projectile.GetDamageAmount(CE_Utility.GetWeaponFromLauncher(launcher));
                 DamageDefExtensionCE damDefCE = def.projectile.damageDef.GetModExtension<DamageDefExtensionCE>() ?? new DamageDefExtensionCE();
+                float armorPenetration = def.projectile.GetArmorPenetration(CE_Utility.GetWeaponFromLauncher(launcher));
```

Comment:
This should be taken from projectilePropsCE, vanilla method will return bogus values.

Reply:
Okay will use that variable.